rubix-core/src/main/java/com/qubole/rubix/core/NonLocalRequestChain.java

```diff
@@ -165,4 +166,45 @@ public Integer call() throws Exception
 
     return nonLocalReadBytes;
   }
+
+  @Override
+  public void updateCacheStatus(String remotePath, long fileSize, long lastModified, int blockSize, Configuration conf)
+  {
+    if (CacheConfig.isDummyModeEnabled(conf)) {
+      if (remoteFetchRequestChain == null || remoteFetchRequestChain.getReadRequests().isEmpty()) {
```

Comment:
We should just call remoteFetchRequestChain.updateCacheStatus for each of the remoteFetch request chains. This will get rid of the duplication of the code here.